src/main/java/nwoolcan/utils/Result.java

```diff
@@ -167,13 +167,11 @@ public T orElse(final Supplier<? extends T> other) {
      * the provided one.
      * @param exception the type of exception.
      * @param function the function to apply to the exception
-     * @param <E> the type of the exception
      * @return this
      */
-    @SuppressWarnings("unchecked")
-    public <E extends Exception> Result<T> peekError(final Class<E> exception, final Consumer<E> function) {
+    public Result<T> peekError(final Class<? extends Exception> exception, final Consumer<? super Exception> function) {
```

Comment:
I don't like this because there's no direct relation between `Class` generic and `Consumer` one and it forces to use only methods that are present in `Exception` class or above. For instance, `FileSystemException` has additional methods that could not be reached if using that signature. In the previous signature instead, the `Class` and `Consumer` are required to be of the same type, preventing misuse and allowing access to methods of particular instances of `Exception`